apps/experiments/services.py

```diff
@@ -0,0 +1,33 @@
+from django.db import transaction
+from experiments.models import Experiment, ExperimentVersion
+from experiments.serializers import populate_experiment_version_data
+
+
+def save_experiment_version(experiment: Experiment, is_default: bool = False) -> str:
+    version_data = populate_experiment_version_data(experiment)
+    if version_data:
+        last_version = ExperimentVersion.objects.filter(experiment=experiment).order_by("-version_number").first()
+        next_version_number = last_version.version_number + 1 if last_version else 1
```

Comment:
Just something to chew on, feel free to resolve once read: Since we have a unique constraint on `experiment` and `version_number` this could potentiall cause an `IntegrityError` to be raised when someone else also tries to make a new version at the same time. Granted, it's unlikely, but not impossible.

I am not sure what the best way to handle this would be. Do we let it die and give the user a nice message, or do we somehow figure out that the `version_number` was the one that caused this, so we simply retry by incrementing the number that we wanted to add originally.